src/ion/requirement.rs

```diff
@@ -31,39 +32,53 @@ impl Requirement {
             (other, Requirement::Any) | (Requirement::Any, other) => other,
-            (Requirement::Stack, Requirement::Stack) => self,
-            (Requirement::Register, Requirement::Fixed(preg))
-            | (Requirement::Fixed(preg), Requirement::Register) => Requirement::Fixed(preg),
             (Requirement::Register, Requirement::Register) => self,
-            (Requirement::Fixed(a), Requirement::Fixed(b)) if a == b => self,
+            (Requirement::Stack, Requirement::Stack) => self,
+            (Requirement::Register, Requirement::FixedReg(preg))
+            | (Requirement::FixedReg(preg), Requirement::Register) => Requirement::FixedReg(preg),
+            (Requirement::Stack, Requirement::FixedStack(preg))
+            | (Requirement::FixedStack(preg), Requirement::Stack) => Requirement::FixedStack(preg),
+            (Requirement::FixedReg(a), Requirement::FixedReg(b)) if a == b => self,
+            (Requirement::FixedStack(a), Requirement::FixedStack(b)) if a == b => self,
             _ => Requirement::Conflict,
         }
     }
+}
+
+impl<'a, F: Function> Env<'a, F> {
     #[inline(always)]
-    pub fn from_operand(op: Operand) -> Requirement {
+    pub fn requirement_from_operand(&self, op: Operand) -> Requirement {
         match op.constraint() {
-            OperandConstraint::FixedReg(preg) => Requirement::Fixed(preg),
+            OperandConstraint::FixedReg(preg) => {
+                if self.pregs[preg.index()].is_stack {
+                    Requirement::FixedStack(preg)
+                } else {
+                    Requirement::FixedReg(preg)
+                }
+            }
             OperandConstraint::Reg | OperandConstraint::Reuse(_) => Requirement::Register,
             OperandConstraint::Stack => Requirement::Stack,
             _ => Requirement::Any,
         }
     }
-}
 
-impl<'a, F: Function> Env<'a, F> {
-    pub fn compute_requirement(&self, bundle: LiveBundleIndex) -> Requirement {
+    pub fn compute_requirement(&self, bundle: LiveBundleIndex) -> (Requirement, ProgPoint) {
         let mut req = Requirement::Unknown;
         log::trace!("compute_requirement: {:?}", bundle);
-        for entry in &self.bundles[bundle.index()].ranges {
+        let ranges = &self.bundles[bundle.index()].ranges;
+        for entry in ranges {
             log::trace!(" -> LR {:?}", entry.index);
             for u in &self.ranges[entry.index.index()].uses {
                 log::trace!("  -> use {:?}", u);
-                let r = Requirement::from_operand(u.operand);
+                let r = self.requirement_from_operand(u.operand);
                 req = req.merge(r);
                 log::trace!("     -> req {:?}", req);
+                if req == Requirement::Conflict {
+                    return (req, u.pos);
+                }
             }
         }
         log::trace!(" -> final: {:?}", req);
-        req
+        (req, ranges.first().unwrap().range.from)
```

Comment:
Perhaps I slightly misunderstand the purpose of the `ProgPoint`, given this -- in this case we don't need a split, so I guess we can return `None` (if `Option` as I suggested above)?